conjur/user/user_logic.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+
+"""
+UserLogic module
+
+This module is the business logic for handling all user-related activity
+"""
+
+# Builtins
+import logging
+
+# Internals
+from conjur.credentials_data import CredentialsData
+
+class UserLogic:
+    """
+    UserLogic
+
+    This class holds the business logic for handling
+    user activity
+    """
+    def __init__(self, conjurrc_data, credentials_from_store, client, resource):
+        self.conjurrc_data=conjurrc_data
+        self.credentials_from_store=credentials_from_store
+        self.client=client
+        self.resource=resource
+
+    def rotate_api_key(self, resource_to_update):
+        """
+        Method to trigger two types of API key actions.
+        1. The user can either rotate their own API key or
+        2. Rotate another users API key
+        """
+        if resource_to_update is not None:
+            logging.debug(f"Rotating API key for '{resource_to_update}'")
+            new_api_key = self.rotate_anothers_api_key(resource_to_update)
+            logging.debug(f"API key for '{resource_to_update}' successfully rotated")
+        else:
```

Comment:
From UX perspective are we sure we want to rotate if the user dont specify anything ? Maybe we want him to specifiy it implicitly? Because if he use it without flags maybe he won't want to rotate his own api key?

Reply:
This is the same ux as in Ruby but it is question for @sharonr78 why we want to keep this.

Reply:
Hi @sigalsax and @tzheleznyak,
The suggested UX is to prompt for a username or host to rotate their key, but when used without flags than it is implicit.
For instance:
`conjur user rotate-api-key -i joe` will rotate the key for user joe, while,
`conjur user rotate-api-key` will prompt the logged-in user to specify a name for api key rotation. It can be his own name, or another name.
- Checkout this commands mapping for a reference: https://ljfz3b.axshare.com/#id=c63xjv&p=commands_matrix___rules&g=1

@InbalZilberman please confirm

Reply:
These described flows are what is being implemented. @tzheleznyak  is suggesting that `conjur use rotate-api-key` is too intrusive (correct me if I am wrong Tamir) and a user could run `conjur user rotate-api-key` and expect to see the help screen for example for further guidance